### PR TITLE
Allows Other Components and Extensions to Modify Statement

### PR DESCRIPTION
**Pull request by bbeversdorf**
Trigger an event before and after sending a statement to the LRS to allow other plugins the opportunity to modify the statement.

### CODE REVIEW

**Comment by brian-learningpool** [comment]
Hi @bbeversdorf, can you provide a use-case for this?  Please feel free to join us on the Gitter chat:
https://gitter.im/adaptlearning/xAPI

**Comment by bbeversdorf** [comment]
@brian-learningpool Our use case deals with adding an "other" context activity to attach more information to a statement. Specifically when the user completes a course, we want the statement context to indicate a related assessment so the user could test their knowledge in the course.

 To do this we created an extension to listen to this xAPI Plugin before the statement is sent. Then the extension modifies the statement accordingly. 
```...
"context": {
		"contextActivities": {
			"other": [{
				"id": ".......activity/e023c45b-ed97-4724-b8c8-2b2fc7b34b99/#/id/592861893e01741e2028336e",
				"objectType": "Activity",
				"definition": {
					"name": {
						"en-US": "Post-assessment"
					},
					"type": "http://adlnet.gov/expapi/activities/assessment",
					"extensions": {
						"http://id.tincanapi.com/extension/assessment-type": "posttest"
					}
				}
			}]
		},
		"language": "en"
	},
...```

File: js/adapt-xapi.js [hunk 1]
```diff
@@ -416,8 +416,9 @@ define(function(require) {
       if (!statement) {
         return;
       }
-
+      Adapt.trigger("xapi:willSendStatement", statement);
```
Comment (brian-learningpool) [inline js/adapt-xapi.js]: Can you rename these events to `xapi:preSendStatement` and `xapi:postSendStatement` please?  Also, target at the `develop` branch.

Comment (brian-learningpool) [inline js/adapt-xapi.js]: Also, on that branch you'll find an equivalent call to sendStatements which will need similar events.

**Comment by brian-learningpool** [comment]
Hi @bbeversdorf, I haven't heard from you for a while.

I've just added support for `preSendStatement` and `preSendStatements` events to the `develop` branch.

**Pull request closed without merging**